contracts/strategy/yearn/YearnStrategy.sol

```diff
@@ -135,7 +136,9 @@ contract YearnStrategy is IStrategy, AccessControl, CustomErrors {
         }
 
         // transfer underlying to vault
-        underlying.safeTransfer(vault, amount);
+
+        underlying.safeIncreaseAllowance(address(this), amount);
```

Comment:
can't we just do a underlying.safeIncreaseAllowance MAX in the constructor and go with it? this is just transferring money between contracts cc @naps62

Reply:
I suppose we can do this. I will go forward and make the change

Reply:
[changed](https://github.com/lindy-labs/sc_solidity-contracts/pull/139/commits/7c1b86a53a9692aad901d0ba9640ddbd6d0d0a49)